packages/sol6/src/yang/etsi-nfv-ns.yang

```diff
@@ -390,64 +426,113 @@ submodule etsi-nfv-ns {
           }
           description
             "References the Virtual Link Profile of a constituent
-             VL.";
+            VL.";
           reference
             "GS NFV IFA014: Section 6.4.2.2 Vnffgd information
-             element";
+            element";
         }
 
-        list cpd-pool-id {
+        list cpd-pool {
           // This is supposed to be a reference to a CpdPool.
           // However, it is not clear where CpdPool is maintained.
           // For now it is maintained here.
           key "id";
-          
+
           description
             "A reference to a pool of descriptors of connection
-             points attached to one of the constituent VNFs and PNFs
-             and/or one of the SAPs of the parent NS or of a nested
-             NS.";
+            points attached to one of the constituent VNFs and PNFs
+            and/or one of the SAPs of the parent NS or of a nested
+            NS.";
           reference
             "GS NFV IFA014: Section 6.4.2.2 Vnffgd information
-             element";
+            element";
 
           leaf id {
             type string;
             description
               "Identifier of this CpdPool information element. It
-               identifies a pool of descriptors of connection points
-               and NS SAPs.";
+            identifies a pool of descriptors of connection points
+            and NS SAPs.";
             reference
               "GS NFV IFA014: Section 6.4.4.2 CpdPool information
-               element";
-          }
-
-	  choice cpd-id {
-	    leaf vnf-cpd-id {
-	      type leafref {
-		path "deref(../../vnf-profile-id)/../" +
-		  "virtual-link-connectivity/cpd-id";
-	      }
-	    }
-	    leaf pnf-cpd-id {
-	      type leafref {
-		path "deref(../../pnf-profile-id)/../" +
-		  "virtual-link-connectivity/cpd-id";
-	      }
-	    }
-	    leaf sap-cpd-id {
-	      type leafref {
-		path "deref(../../nested-ns-profile-id)/../" +
-		  "virtual-link-connectivity/cpd-id";
-	      }
-	    }
+            element";
+          }
+
+          leaf df-id {
```

Comment:
This is not in the IFA spec. If we need this, we will need to request for change. Let us discuss this in the call.

Reply:
Yes!

Reply:
Also, the original intention of keeping track of CpdPool was to keep track of CP that are participating in the VNFFGD. This change assumes that all CP associated with a VNFD, pointed to by the VNF Profile, also belong to the VNFFGD. Can that be assumed?